### ISSUE TITLE
Add RegEx for Hiding Branch Names

### ISSUE THREAD

**Issue by brianwallace** [gladed/gradle-android-git-version#23]
In some cases it would be nice to be able to hide all branch names for hide branch names for s set of changes.  Adding RegEx support to the `hideBranch` field would make this flexible and possible.

Examples:

```
// Hide ALL branch names
androidGitVersion {
    hideBranches = [ '*' ]
}
```

```
// Hide branch names in 'master', 'release', 'develop' and all 'feature/*' branches
androidGitVersion {
    hideBranches = [ 'master', 'release', 'develop', 'feature/*' ]
}
```

It looks like this [bit of code](https://github.com/gladed/gradle-android-git-version/blob/90a8d4ca6749f381ccc2bf16c89a36701fd0f27d/src/main/groovy/com/gladed/gradle/androidgitversion/AndroidGitVersion.groovy#L92) would need tweaking:

```
if (!hideBranches.contains(results.branchName)) {
    name += "-" + results.branchName.replaceAll("[^a-zA-Z0-9.-]", "_")
}
```

**Comment by GladeDiviney** [comment]
Good idea. If it's regex it would need to be e.g. `feature/.*`

**Comment by gladed** [comment]
Fixed in 1c256ee (0.2.19)
